main.go

```diff
@@ -40,8 +41,22 @@ func main() {
 	buf := bufio.NewReader(os.Stdin)
 	var regArr []*regexp.Regexp
 
-	if !*fixedStringsParameter {
+	if *wordsParameter && !*fixedStringsParameter {
+		for i := range flagTest {
+			if !strings.HasPrefix(flagTest[i], "\\b") {
+				flagTest[i] = "\\b" + flagTest[i]
+			}
+			if !strings.HasSuffix(flagTest[i], "\\b") {
+				flagTest[i] = flagTest[i] + "\\b"
+			}
+		}
+	}
+
+	if !*fixedStringsParameter || *wordsParameter && *fixedStringsParameter {
 		for _, i := range flagTest {
+			if *wordsParameter && *fixedStringsParameter {
+				i = regexp.QuoteMeta(i)
```

Comment:
Any way to reduce the whole code's -F handling independent of -w to this one in a dedicated loop?

Reply:
if you say so there probably is one but i dont quite get it